src/components/calcite-list-item/calcite-list-item.tsx

```diff
@@ -7,6 +7,7 @@ import { getSlotted } from "../../utils/dom";
  * @slot content-start - A slot for adding non-actionable elements before the label and description of the list item.
  * @slot content-end - A slot for adding non-actionable elements after the label and description of the list item.
  * @slot actions-end - A slot for adding actionable `calcite-action` elements after the content of the list item.
+ * @slot - A slot for adding `calcite-list-item` and `calcite-list-item-group` elements.
```

Comment:
@Esri/calcite-design-representatives This PR introduces list item grouping. Design-wise, do we need to add a note on the max amount of nesting to support?

Reply:
That would be good to add. 👍